modules/emulator/R/GaussProcess.R

```diff
@@ -165,119 +185,131 @@ function(x,y,isotropic=TRUE,nugget=TRUE,method="bayes",ngibbs=5000,burnin=1000,t
-    if(prior == "IG"){
-      anum <- ldinvgamma(taustar,aw,bw)
-      aden <- ldinvgamma(tauw,aw,bw)
+    ## draw tauw
+    taustar <- exp(rnorm(1, log(tauw), p(tauwjump)))
+    Sstar   <- calcSpatialCov(d, psi, taustar)
+    anum    <- aden <- 0
+    if (prior == "IG") {
+      anum <- ldinvgamma(taustar, aw, bw)
+      aden <- ldinvgamma(tauw, aw, bw)
     }
-    anum <- try( anum + dmvnorm(as.vector(W),rep(0,n.unique),Sstar,log=TRUE))
-    aden <- aden + dmvnorm(as.vector(W),rep(0,n.unique),S,log=TRUE)
-    if(is.numeric(anum) && is.finite(anum)&& exp(anum-aden) > runif(1)) {
+    anum <- try(anum + dmvnorm(as.vector(W), rep(0, n.unique), Sstar, log = TRUE))
+    aden <- aden + dmvnorm(as.vector(W), rep(0, n.unique), S, log = TRUE)
+    if (is.numeric(anum) && 
+        is.finite(anum) && 
+        exp(anum - aden) > runif(1)) {
       tauw <- taustar
       S <- Sstar
     }
-    tauwgibbs[g,] <- tauw
-    tauwjump <- update(tauwjump,tauwgibbs)
+    tauwgibbs[g, ] <- tauw
+    tauwjump <- update(tauwjump, tauwgibbs)
     
     cc <- 4
-    ##draw tauv
-    if(nugget){
-      tauv <- rinvgamma(1,av + n/2, bv + 0.5*sum((y-rep(mu,n)-W.full)^2))
-      Tinv <- diag(1/tauv,n)
+    ## draw tauv
+    if (nugget) {
+      tauv <- rinvgamma(1, av + n / 2, bv + 0.5 * sum((y - rep(mu, n) - W.full) ^ 2))
+      Tinv <- diag(1 / tauv, n)
     }
-  
+    
     cc <- 5
-    ##draw mu
-    if(zeroMean){
+    ## draw mu
+    if (zeroMean) {
       mu <- 0
     } else {
-      if(nugget){
-        M <- 1/(n/tauv+1/mu.V0)
-        m <- sum(y-W.full)/tauv
-        mu <- rnorm(1,M*m,M)
+      if (nugget) {
+        M <- 1 / (n / tauv + 1 / mu.V0)
+        m <- sum(y - W.full) / tauv
+        mu <- rnorm(1, M * m, M)
       } else {
         Sinv <- solve(S)
-        M <- solve(t(X) %*% Sinv %*% X + 1/mu.V0)
-        m <- t(X) %*% Sinv %*% y ##[1:10]
-        mu <- rnorm(1,M*m,M)
+        M <- solve(t(X) %*% Sinv %*% X + 1 / mu.V0)
+        m <- t(X) %*% Sinv %*% y  ##[1:10]
+        mu <- rnorm(1, M * m, M)
       }
     }
     
     cc <- 6
-    ##store
-    if(g  %in% samp){
+    ## store
+    if (g %in% samp) {
       i <- which(samp == g)
       mugibbs[i] <- mu
-      ## psigibbs[i] <- psi
-      ## tauwgibbs[i] <- tauw
-      if(nugget){
+      ## psigibbs[i] <- psi tauwgibbs[i] <- tauw
+      if (nugget) {
         tauvgibbs[i] <- tauv
-        Wgibbs[i,] <- W
+        Wgibbs[i, ] <- W
       }
     }
-    if(haveTime) prevTime <- progressBar(g/ngibbs,prevTime)
-
+    if (haveTime) {
+      prevTime <- progressBar(g/ngibbs, prevTime)
+    }
+  }
+  if (haveTime) {
+    progressBar(1.1, prevTime)
   }
-  if(haveTime) progressBar(1.1,prevTime);
   
-  return(list(method=method,tauwjump=tauwjump,tauw = tauwgibbs,psijump=psijump,psi=psigibbs,mu=mugibbs,tauv=tauvgibbs,W=Wgibbs,nugget=nugget,isotropic=isotropic,d=d,samp=samp,x.id=x.id,x.compact=x.compact,y=y,zeroMean=zeroMean))
-}
-
+  list(method = method, tauwjump = tauwjump, tauw = tauwgibbs, 
```

Comment:
@istfer Reason - just that I feel it's more "R style". But see my other comment; happy to leave them in if that's desired.